packages/contracts/contracts/test/TestSharedComponent.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "../core/component/Component.sol";
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+/// @notice A test component that manages permission to internal objects by associating their IDs with specific DAOs. Only the DAO for which the object was created has the permission to perform ID-gated actions on them.
+/// @dev This is realized by asking an `IACLOracle` that must be authorized in the DAO's ACL.
+contract TestSharedComponent is Component {
+    bytes32 public constant ID_GATED_ACTION_ROLE = keccak256("ID_GATED_ACTION_ROLE");
+
+    uint256 internal _counter;
+
+    modifier sharedAuth(uint256 _id, bytes32 _role) {
+        if (!ownedIds[_id].hasPermission(address(this), _msgSender(), _role, _msgData()))
+            revert ACLData.ACLAuth({here: address(this), where: address(this), who: _msgSender(), role: _role});
```

Comment:
add curly brackets for better readability

Reply:
done